### PR TITLE
Update debug info emission to handle funclets

### PR DESCRIPTION
**Pull request by JosephTremoulet**
The object gets a function symbol for each funclet, so we need to sum
their sizes rather than trying to report each in turn.

### CODE REVIEW

**Comment by JosephTremoulet** [comment]
Note: this is a PR for a change into the EH branch.  With this change and Microsoft/llvm#98, we can successfully execute simple try/catches using the EH branch with no workarounds.

@AndyAyersMS or @adiaaida PTAL.

File: lib/Jit/LLILCJit.cpp [hunk 1]
```diff
@@ -560,69 +565,74 @@ void ObjectLoadListener::getDebugInfoForObject(
...
+      // The main function is always laid out first
+      Addr = SingleAddr;
+    }
+    Size += SingleSize;
+  }
 
-    uint32_t LastDebugOffset = (uint32_t)-1;
-    uint32_t NumDebugRanges = 0;
-    ICorDebugInfo::OffsetMapping *OM;
+  uint32_t LastDebugOffset = (uint32_t)-1;
```
Comment (JosephTremoulet) [inline lib/Jit/LLILCJit.cpp]: I don't know why the diff isn't figuring this out, but fyi everything from here down is just an indentation change (and removing the closing brace where there used to be a loop).

**Comment by AndyAyersMS** [comment]
LGTM

**Comment by adiaaida** [comment]
LGTM too

**Pull request merged**